processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -900,6 +927,27 @@ public Rowboat apply(@Nullable Rowboat input)
         )
     );
 
+    if (segmentMetadata != null && !segmentMetadata.isEmpty()) {
+      File metadataFile = new File(v8OutDir, "metadata.drd");
+      FileOutputStream metadataFileOutputStream = null;
+      FileChannel metadataFilechannel = null;
+      try {
+        metadataFileOutputStream = new FileOutputStream(metadataFile);
+        metadataFilechannel = metadataFileOutputStream.getChannel();
+        metadataFilechannel.write(ByteBuffer.wrap(mapper.writeValueAsBytes(segmentMetadata)));
+      }
+      finally {
+        CloseQuietly.close(metadataFilechannel);
```

Comment:
I followed th same pattern as for other files inside the segment e.g. index.drd , see https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/segment/IndexMerger.java#L521 
if we want to change any of that then it needs to thought/changed in all places and I wouldn't want to do the in this PR really.